src/java.base/share/classes/jdk/internal/vm/vector/VectorSupport.java

```diff
@@ -621,6 +625,26 @@ VOUT convert(int oprId,
 
     /* ============================================================================ */
 
+    public interface ComExpOperation<V extends Vector<?>,
+                                     M extends VectorMask<?>> {
+        V apply(V v1, V v2, M m);
+    }
+
+    @IntrinsicCandidate
+    public static
+    <V extends Vector<E>,
+     M extends VectorMask<E>,
+     E>
+    V comExpOp(int opr,
+              Class<? extends V> vClass, Class<? extends M> mClass, Class<E> eClass,
+              int length, V v1, V v2, M m,
```

Comment:
No need to use two vectors here. Use zeroing-masking with one vector could meet requirement, right?

Reply:
Two vectors are added in anticipation to support an explicit background vector in future since original APIs had them. 
From the Java side background vector passed currently is ZERO which is why a merging semantics is used in emitted compess/expand instruction instead of zeroing semantics.